%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% I, the copyright holder of this work, release this work into the
%% public domain. This applies worldwide. In some countries this may
%% not be legally possible; if so: I grant anyone the right to use
%% this work for any purpose, without any conditions, unless such
%% conditions are required by law.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\documentclass{beamer}
\setbeamertemplate{caption}[numbered]


\usetheme[faculty=ped]{fibeamer}
\usepackage[utf8]{inputenc}
\usepackage[main=english, czech, slovak]{babel}
\makeatletter % remove the FACULTY OF ... MASARYK UNI LOGO
\renewcommand\fibeamer@includeLogo[1][]{}
\makeatother
\title{Phylogenetics for Predicting Virus Evolution}
\subtitle{Anticipating next seasons influenza strains}
\author{Paul Wiesemeyer}
\usepackage{ragged2e}  % `\justifying` text
\usepackage{booktabs}  % Tables
\usepackage{tabularx}
\usepackage{tikz}      % Diagrams
\usetikzlibrary{calc, shapes, shapes.geometric, arrows, backgrounds}
\tikzstyle{startstop} = [rectangle, rounded corners, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=orange]
\usepackage{amsmath, amssymb}

\usepackage{hyperref}       % `\url`s
\usepackage{listings}  % Code listings
\frenchspacing

%\usepackage[backend=biber,style=alphabetic,citestyle=authoryear]{biblatex}
%\addbibresource{virophyle.bib}

%      \bibliographystyle{unsrt}

%      \bibliography{virophyle}

%\usepackage[plainnat]{natbib}



\begin{document}
  \shorthandoff{-}
  \frame{\maketitle}

    % \AtBeginSection[]{% Print an outline at the beginning of sections
  
    %     \begin{frame}<beamer>
    %       \frametitle{Outline for Section \thesection}
    %       \tableofcontents[currentsection]
    %     \end{frame}
    % }
    
  \begin{darkframes}




  \begin{frame}<beamer>
    \frametitle{Problem}

    \begin{tikzpicture}[remember picture,overlay]
      \node[xshift=-4cm,yshift=-2cm] at (current page.north east) {\includegraphics[width=.7\textwidth]{resources/Houston.pdf}};
    \end{tikzpicture}

    \begin{itemize}
      \itemsep1em
      \item Estimated $3 - 11 \% $ of the population catch symptomatic influenza each season \cite{tokarsOlsen+18a}
      \item Influenza evolves rapidly and evades immunity and vaccines
      \item Epidemics and pandemics caused by RNA recombination events
      \item Science is slow responding to health issues
    \end{itemize}
  \end{frame}

  \begin{frame}<beamer>
    \frametitle{Problem}
    \begin{itemize}
      \item $\sim$ \textit{8 months} from submission to publication of a medical paper \cite{aAMC18}
      \item $\sim$ \textit{6 months} from vaccine strain selection to distribution:
    \end{itemize}
    \begin{figure}
        \includegraphics[width=\textwidth]{resources/flu_vaccine_schedule.png}
        \caption{\cite{bedford15}}
    \end{figure}{}
  \end{frame}





  \begin{frame}<beamer>
           \frametitle{Outline}
           \footnotesize
           \tableofcontents
  \end{frame}







  \section{Influenza}

    \begin{frame}{\secname}
      \framesubtitle{a closer look}

      \begin{tikzpicture}
        \usebeamercolor{fibeamer}
        \node [anchor=west] (rna) at (-1,6.5) {\footnotesize 8 single stranded RNA};
        \node [anchor=west] (ha) at (-1,1) {\footnotesize Hemagglutinin (HA)};
        \node [anchor=west] (na) at (-1,3) {\footnotesize Neuraminidase (NA)};
        \node [anchor=west] (cdc) at (-1,0) {\footnotesize Image from \cite{cDC20}};
        \begin{scope}[xshift=1.5cm]
          \node[anchor=south west,inner sep=0] (image) at (0,0) {\includegraphics[width=0.8\textwidth]{resources/flu-JPG.png}};
          \begin{scope}[x={(image.south east)},y={(image.north west)}]
            \draw [-latex, thick, orange] (rna) to[out=0, in=150] (0.48,0.90);
            \draw [-latex, thick, orange] (ha) -- ++(0.64,0.0);
            \draw [-latex, thick, orange] (na) -- ++(0.34,0.0);
          \end{scope}
        \end{scope}
      \end{tikzpicture}

    \end{frame}


    \begin{frame}{\secname}
      \framesubtitle{jumps far}

      \begin{figure}
        \includegraphics[width=\textwidth]{resources/alberts2015molecularp1292.pdf}
        \caption{Influenza A evolutionary shift events \cite{alberts15}}
      \end{figure}{}
    \end{frame}{}


    \begin{frame}{\secname}
      \framesubtitle{runs fast}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{resources/h3n2escape.pdf}
        \caption{\footnotesize Phylogeny of Influenza A/H3N2 Hemagglutinin \cite{volzKoelle+13}}
      \end{figure}
    \end{frame}

    % \begin{frame}{\secname}
    %   \framesubtitle{A virus that does not change escape vehicle will get caught}
    %   \begin{figure}
    %     \includegraphics[width=.9\textwidth]{resources/GTAsa.jpg}
    %     \caption{\footnotesize It's like getting more and more stars, having to switch escape vehicle constantly}
    %   \end{figure}
    % \end{frame}







  \section{Phylogenetics}

    \begin{frame}{\secname}
      \framesubtitle{Terminology}

      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-1cm,yshift=-1.7cm] at (current page.north east) {\includegraphics[width=\textwidth]{resources/Haeckel.pdf}};
        \node [anchor=west] (cdc) at (-1,-5.5) {\footnotesize Image from \cite{haeckel66}};
      \end{tikzpicture}

      Tree---a simply connected graph

      \begin{itemize}
        \item branches

        \item nodes (vertices)

        \item leaves (endpoints)

        \item root $\Rightarrow$ parents, children

        \item clade---a collection of branches

      \end{itemize}
    \end{frame}



    \begin{frame}{\secname}

      \framesubtitle{Concept: Parsimony}

      \begin{figure}
        \includegraphics[width=.7\textwidth]{resources/parsimony2.pdf}
        \caption{\footnotesize Reconstructing a tree requires likelihood considerations}
      \end{figure}
      \footnotesize

      Minimize the number of mutations to get the most likely tree(s)\\[1mm]

      Problem: for $n$ leaves, \# possible trees $\propto 2^n\ n !$\\[1mm]

      $\Rightarrow$ We can always just compute a small number of trees

    \end{frame}

    \subsection{Sequence Alignment}

    \begin{frame}{\secname}
      \framesubtitle{Example algorithm: \subsecname}

      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-2.2cm,yshift=-2.2cm] at (current page.north east) {\includegraphics[width=.4\textwidth]{resources/align.pdf}};
      \end{tikzpicture}
      \vfill
      N sequences of lengths $n_1 ... n_N$ (not same due to indels, seq. errors)
      \begin{itemize}
        \item Pairwise matching, give a penalty, make N x N table
        \item Lowest penalties group together, form new subunit.
        \item Iterate. When comparing to sets of sequences, use mean.
      \end{itemize}

      Result: small number of locally most likely trees, not necessarily global maximum.

      Effectively in use today: Markov Chain Monte Carlo "climber"

    \end{frame}


    \subsection{The Molecular Clock}

    \begin{frame}{\secname}
      \framesubtitle{Concept: \subsecname}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/molClock1.pdf}
        \caption{\footnotesize Linear time-mutation relationship}
      \end{figure}
      \addtocounter{figure}{-1}
    \end{frame}


    \begin{frame}{\secname}
      \framesubtitle{Concept: \subsecname}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/molClock2.pdf}
        \caption{\footnotesize Linear time-mutation relationship}
      \end{figure}
      \vspace*{-.2cm}
      \footnotesize
      Strongly simplified. Different genome sites experience different evolutionary pressure. Say: Silent vs. non-silent, each site in genome can have a different rate. Then it is hard to deduce anything, possible intermediate: Distribution of rates.
    \end{frame}





  \section{Predicting the next strain of influenza}



    \subsection{Predicting Virus Evolution}

    \begin{frame}{\secname}
      \framesubtitle{Approaches}

      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-2.9cm,yshift=-3.4cm] at (current page.north east) {\includegraphics[width=.5\textwidth]{resources/koel7.pdf}};
        \node[xshift=3.9cm,yshift=-3.5cm] at (current page.north west) {\includegraphics[width=.5\textwidth]{resources/map.png}};
        \node [anchor=west] (cdc) at (-1,-5.5) {\footnotesize Map (symbolic) from \cite{russellJones+08a}, H3N2 HA epitope site from \cite{koelBurke+13}};

      \end{tikzpicture}

      \vfill
      Epidemiology: Geographical region?\\[2mm]

      Molecular Biology: Is there a certain type of Mutation?\\[2mm]

      Immunology: Hemagglutinin inhibition cartography\\[2mm]

      \Large Vision: Bring these levels together
    \end{frame}




    \subsection{The Hemagglutination Inhibition Assay}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{resources/hemagglutination.pdf}
        \caption{Red blood cells (RBC) precipitate.}
        \label{1}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{resources/hemagglutinationAssay.pdf}
        \caption{Influenza Hemagglutinin (HA) coagulates the RBC, forming a mat.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{resources/hemagglutinationInhibitionAssay.pdf}
        \caption{Antisera that fit the HA's epitope site bind it, letting the RBC sink to the bottom. Effect works up to a certain antigenic distance and .}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.9\textwidth]{resources/HIassaylab.jpg}
        \caption{\footnotesize Here, one antiserum is tested in 12 different dilutions against 8 different virus strains. The highest dilution that prevents agglutination is called the titer.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{How this used to be looked at}
      % \begin{figure}
      %   \centering
      %   \begin{subfigure}{.5\textwidth}
      %     \centering
      %     \includegraphics[width=\linewidth]{resources/HItable.png}
      %     \caption{A subfigure}
      %     \label{fig:sub1}
      %   \end{subfigure}%
      %   \begin{subfigure}{.5\textwidth}
      %     \centering
      %     \includegraphics[width=.4\linewidth]{resources/hi_smith_map.jpg}
      %     \caption{A subfigure}
      %     \label{fig:sub2}
      %   \end{subfigure}
      %   \caption{A figure with two subfigures}
      %   \label{fig:test}
      %   \end{figure}
    \end{frame}

    \subsection{Mapping Antigenicity to the Tree}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/HItableToTree.pdf}
        \caption{\footnotesize Mapping the chart to the tree constructed from sequences}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/HItableToTreeDiffs.pdf}
        \caption{\footnotesize Here, one antiserum is tested in 12 different dilutions against 8 different virus strains. The highest dilution that prevents agglutination is called the titer.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/HItableToTreeUru.pdf}
        \caption{\footnotesize Here, one antiserum is tested in 12 different dilutions against 8 different virus strains. The highest dilution that prevents agglutination is called the titer.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{\small The asymmetry makes sense, think of it like this but with more dimensions}
      \begin{figure}
        \includegraphics[width=.58\textwidth]{resources/shapesOfWood.jpg}
        \caption{\footnotesize \cite{rosipaw10}}
      \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]{Formulas}
      \framesubtitle{}
      $T_{a\beta}~\ $........................... HI titer of virus $a$ against antiserum $\beta$ (virus $b$)

      $H_{a\beta}~$........................... $\log_2$ relative titer (we'll use this one)
      \begin{align}
        H_{a\beta} = \log_2 (T_{b\beta}) ~ - ~\log_2 ( T_{a\beta})
      \end{align}

      $\hat{H}_{a\beta}~$........................... predicted $\log_2$ relative titer
      $v_{a}~~~$........................... avidity of virus $a$ (=greediness)
      $p_{\beta}~~~$........................... potency of antiserum $\beta$ (=effectiveness)
      $D_{a\beta}~$........................... genetic component of titer drop

      \begin{align}
        \hat{H}_{a\beta} = v_a + p_\beta + D_{ab} \text{   D is between two viruses}
      \end{align}


      What remains is split up into a sum over individual mutation contributions:
      \begin{align}
        D_{ab} = \sum_{i \in (a ... b)} d_i
      \end{align}
      Where the sum is over the path connecting virus $a$ and virus $ b \stackrel{}={}$ antiserum $\beta$. Now we want
      \begin{align}
        \hat{H}_{a\beta} \stackrel{!}={} {H}_{a\beta}
      \end{align}
      to that end we minimize a cost function $C$ of the whole tree:
      \begin{align}
        C := \sum_{a,\beta}( \hat{H}_{a\beta} - {H}_{a\beta})^2 + \lambda \sum_i d_i + \gamma \sum_a v_a^2 + \delta \sum_\alpha p_\alpha^2
      \end{align}

    \end{frame}














  \subsection{Results}



    \begin{frame}{\subsecname}
      \framesubtitle{Omitting 10\% of the HI titer data, to make a test}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/neher+16/tree_model.pdf}
        \caption{\footnotesize On the left, 10\% randomly picked measurements were omitted,\

        on the right, 10 \% of entire titer columns were held back, as if it was a new clade}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{Asymmetry not interesting, but maybe quartet rule?}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/neher+16/tree_model_hist.pdf}
        \caption{\footnotesize a}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{Does recent antigenic evolution have fixation implications?}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/neher+16/recent_antigenic_evolution.pdf}
        \caption{\footnotesize Fraction of samples having a certain mutation plotted vs. time

        Strains with frequencies smaller than $0.01$ were omitted.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{thresholds}
      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-4cm,yshift=-1.5cm] at (current page.north east) {\includegraphics[width=.7\textwidth]{resources/neher+16/recent_antigenic_evolution.pdf}};
      \end{tikzpicture}
      \begin{figure}
        \includegraphics[width=.65\textwidth]{resources/h3n2escape.pdf}
        \caption{\footnotesize GET nextstrain tree with trunk colored according to antigenic advance}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{thresholds}
      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-4cm,yshift=-1.5cm] at (current page.north east) {\includegraphics[width=.7\textwidth]{resources/neher+16/recent_antigenic_evolution.pdf}};
      \end{tikzpicture}
      \begin{figure}
        \includegraphics[width=.8\textwidth]{resources/neher+16/fraction_frequency_threshold_colourbar.pdf}
        \caption{\footnotesize a}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{distance to season year}
      \begin{figure}
        \includegraphics[width=.8\textwidth]{resources/neher+16/distance_to_season_year.pdf}
        \caption{\footnotesize a}
      \end{figure}
    \end{frame}










  \section{\textit{Nextstrain}}

    \begin{frame}{\secname}
      \framesubtitle{Mending pieces together}
      Please visit nextstrain.org/narratives/.....

      Ideas: COVID-Line? H3N2-Line?

      Multiple data sets?

      de Vries Epitope sites

      Koel 7 sites

      D614G --- Epidemiologic turning point
    \end{frame}

    \subsection{How to use the Framework}
    \begin{frame}{\secname : \subsecname}
      \framesubtitle{\href{URL}{https://nextstrain.org/community/narratives/gitchhiker/virophyle}}
      \begin{figure}
        \includegraphics[width=\textwidth]{resources/NextstrainPathogens.pdf}
        \caption{\footnotesize }
      \end{figure}
    \end{frame}

    \subsection{The Powerful Meta Data}
    \begin{frame}{\secname : \subsecname}
      \framesubtitle{(this is mirrored in the narrative)}
    \end{frame}

    \subsection{Confidence Levels and Limitations}
    \begin{frame}{\secname : \subsecname}
      \framesubtitle{(this is mirrored in the narrative)}
    \end{frame}

    \begin{frame}{Multi Scale Evolution}
      \framesubtitle{}
      If a single event mutation occurs, say \textbf{D 186 G} in the HA genome, it is subject to multi scale evolutionary selection:
      \begin{itemize}
        \item this RNA instance vs. the other RNA strands in the same cell
        \item this cell's mutated viruses vs. other viruses inside the host
        \item this hosts viruses vs. viruses in rest of the population
        \item this population vs. other populations
      \end{itemize}
      These scales are difficult to separate. At the population level \textit{epidemiological} processes may dominate.
    \end{frame}

    \begin{frame}{Outlook}
      \framesubtitle{and closing remarks}
      \begin{itemize}
        \item Difficult to disentangle levels, therefore integrate visualization
        \item Include more meta data (symptoms, severity, \& c.)
        \item Interesting: larger evolutionary timescale
        \item Include other pathogens, like bacteria, CAVEAT: HGT
        \item My question: what happens if epidemic threshold is crossed even further, and zoonotic forcing is stronger?
        \item nice to see instance of Red Queen Hypothesis
        \item Transparency and Community building very important!
        \item When you find literature on phylodynamics, often animals, but to me this is more difficult.
      \end{itemize}
    \end{frame}

    \begin{frame}[allowframebreaks]{References}
    \tiny
\bibliography{virophyle}
\bibliographystyle{apalike}
%\printbibliography
    \end{frame}




  \end{darkframes}

\end{document}
