%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% I, the copyright holder of this work, release this work into the
%% public domain. This applies worldwide. In some countries this may
%% not be legally possible; if so: I grant anyone the right to use
%% this work for any purpose, without any conditions, unless such
%% conditions are required by law.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\documentclass{beamer}
\setbeamertemplate{caption}[numbered]


\usetheme[faculty=ped]{fibeamer}
\usepackage[utf8]{inputenc}
\usepackage[main=english, czech, slovak]{babel}
\makeatletter % remove the FACULTY OF ... MASARYK UNI LOGO
\renewcommand\fibeamer@includeLogo[1][]{}
\makeatother
\title{Phylogenetics for Predicting Virus Evolution}
\subtitle{Anticipating next seasons influenza strains}
\author{Paul Wiesemeyer}
\usepackage{ragged2e}  % `\justifying` text
\usepackage{booktabs}  % Tables
\usepackage{tabularx}
\usepackage{tikz}      % Diagrams
\usetikzlibrary{calc, shapes, shapes.geometric, arrows, backgrounds}
\tikzstyle{startstop} = [rectangle, rounded corners, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=orange]
\usepackage{amsmath, amssymb}

\usepackage{hyperref}       % `\url`s
\usepackage{listings}  % Code listings
\frenchspacing

%\usepackage[backend=biber,style=alphabetic,citestyle=authoryear]{biblatex}
%\addbibresource{virophyle.bib}

%      \bibliographystyle{unsrt}

%      \bibliography{virophyle}

%\usepackage[plainnat]{natbib}



\begin{document}
  \shorthandoff{-}
  \frame{\maketitle}

    % \AtBeginSection[]{% Print an outline at the beginning of sections
  
    %     \begin{frame}<beamer>
    %       \frametitle{Outline for Section \thesection}
    %       \tableofcontents[currentsection]
    %     \end{frame}
    % }
    
  \begin{darkframes}




  \begin{frame}<beamer>
    \frametitle{Problem}

    \begin{tikzpicture}[remember picture,overlay]
      \node[xshift=-4cm,yshift=-2cm] at (current page.north east) {\includegraphics[width=.7\textwidth]{linked/Houston.pdf}};
    \end{tikzpicture}

    \begin{itemize}
      \itemsep1em
      \item Estimated $3 - 11 \% $ of the population catch symptomatic influenza each season \cite{tokarsOlsen+18a}
      \item Influenza evolves rapidly, evading immune system recognition
      \item Epidemics and pandemics caused by RNA recombination events
      \item Science is slow responding to health issues
    \end{itemize}
  \end{frame}

  \begin{frame}<beamer>
    \frametitle{Problem}
    \begin{itemize}
      \item {\Large $\sim 8$ \textit{months}} from submission to publication of a medical paper \cite{aAMC18}
      \item {\Large $\sim 6$ \textit{months}} from vaccine strain selection to distribution:
    \end{itemize}
    \begin{figure}
        \includegraphics[width=\textwidth]{linked/flu_vaccine_schedule.png}
        \caption{\footnotesize from \cite{bedford15}}
    \end{figure}{}
    \begin{itemize}
      \item compare: {\LARGE $\sim 1.6$ \textit{days}} doubling time of A/H1N1 (2009) \\
      \footnotesize \cite{mostaco-GuidolinGreer+11}
    \end{itemize}
    
  \end{frame}





  \begin{frame}<beamer>
           \frametitle{Outline}
           \footnotesize
           \tableofcontents
  \end{frame}







  \section{Influenza}

    \begin{frame}{\secname}
      \framesubtitle{a closer look}

      \begin{tikzpicture}
        \usebeamercolor{fibeamer}
        \node [anchor=west] (rna) at (-1,6.5) {\footnotesize 8 single stranded RNA};
        \node [anchor=west] (ha) at (-1,1) {\footnotesize Hemagglutinin (HA)};
        \node [anchor=west] (na) at (-1,3) {\footnotesize Neuraminidase (NA)};
        \node [anchor=west] (cdc) at (-1,0) {\footnotesize Image from \cite{cDC20}};
        \begin{scope}[xshift=1.5cm]
          \node[anchor=south west,inner sep=0] (image) at (0,0) {\includegraphics[width=0.8\textwidth]{linked/flu-JPG.png}};
          \begin{scope}[x={(image.south east)},y={(image.north west)}]
            \draw [-latex, thick, orange] (rna) to[out=0, in=150] (0.48,0.90);
            \draw [-latex, thick, orange] (ha) -- ++(0.64,0.0);
            \draw [-latex, thick, orange] (na) -- ++(0.34,0.0);
          \end{scope}
        \end{scope}
      \end{tikzpicture}

    \end{frame}


    \begin{frame}{\secname}
      \framesubtitle{jumps far}

      \begin{figure}
        \includegraphics[width=\textwidth]{linked/alberts2015molecularp1292.pdf}
        \caption{recent influenza A evolutionary shift events \cite{alberts15}}
      \end{figure}{}
    \end{frame}{}


    \begin{frame}{\secname}
      \framesubtitle{runs fast}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{linked/h3n2escape.pdf}
        \caption{\footnotesize Phylogeny of Influenza A/H3N2 Hemagglutinin \cite{volzKoelle+13}}
      \end{figure}
    \end{frame}

    % \begin{frame}{\secname}
    %   \framesubtitle{A virus that does not change escape vehicle will get caught}
    %   \begin{figure}
    %     \includegraphics[width=.9\textwidth]{linked/GTAsa.jpg}
    %     \caption{\footnotesize It's like getting more and more stars, having to switch escape vehicle constantly}
    %   \end{figure}
    % \end{frame}







  \section{Phylogenetics}

    \begin{frame}{\secname}
      \framesubtitle{Terminology}

      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-1cm,yshift=-1.7cm] at (current page.north east) {\includegraphics[width=\textwidth]{linked/Haeckel.pdf}};
        \node [anchor=west] (cdc) at (-1,-5.5) {\footnotesize Image from \cite{haeckel66}};
      \end{tikzpicture}

      Tree---a simply connected graph

      \begin{itemize}
        \item branches

        \item nodes (vertices)

        \item leaves (endpoints)

        \item root $\Rightarrow$ parents, children

        \item \invisible{root} $\Rightarrow$ clades:\\
        \hspace*{.5cm}a branch with its children

      \end{itemize}
    \end{frame}



    \begin{frame}{\secname}

      \framesubtitle{Concept: Parsimony}

      \begin{figure}
        \includegraphics[width=.7\textwidth]{linked/parsimony2.pdf}
        \caption{\footnotesize Reconstructing a tree requires likelihood considerations}
      \end{figure}
      \footnotesize

      Minimize the number of mutations to get the most likely tree(s)\\[1mm]

      Problem: for $N$ leaves: \hspace*{.5cm}\# possible trees $\propto 2^N\ (N-1) !$\\[1mm]

      $\Rightarrow$ We can usually just compute a minute fraction of these.

    \end{frame}


    \begin{frame}{\secname}
      \framesubtitle{Example algorithm}

      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-2.2cm,yshift=-2.2cm] at (current page.north east) {\includegraphics[width=.4\textwidth]{linked/align.pdf}};
      \end{tikzpicture}
      \vfill
      $N$ sequences of lengths $\ell_1 ... \ell_N$ \small \hspace*{.3cm}(not identical due to indels, seq. errors)
      \begin{itemize}
        \item Compare pairwise, assign a distance, make $N \times N$ table
        \item Lowest distances group together, form new subunit.
        \item Iterate. When comparing to sets of sequences, use arithmetic mean.
      \end{itemize}

      Result: small number of locally most likely trees,\\
      \hspace*{1.06cm}not necessarily global maximum.

      Computing scales with $N^2$

      Better: Markov Chain Monte Carlo

    \end{frame}


    \subsection{The Molecular Clock}

    \begin{frame}{\secname}
      \framesubtitle{Concept: \subsecname}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/molClock1.pdf}
        \caption{\footnotesize Linear time-mutation relationship}
      \end{figure}
      \addtocounter{figure}{-1}
    \end{frame}


    \begin{frame}{\secname}
      \framesubtitle{Concept: \subsecname}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/molClock2.pdf}
        \caption{\footnotesize Linear time-mutation relationship}
      \end{figure}
      \vspace*{-.6cm}
      \footnotesize
      Distinguish between \textit{silent} and \textit{non-silent}---amino acid changing---mutations, the latter are under evolutionary pressure.
    \end{frame}





  \section{Predicting the next strain of influenza}



    \subsection{Approaches}

    \begin{frame}{\secname}
      \framesubtitle{\subsecname}

      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-2.9cm,yshift=-3.4cm] at (current page.north east) {\includegraphics[width=.5\textwidth]{linked/koel7.pdf}};
        \node[xshift=3.9cm,yshift=-3.5cm] at (current page.north west) {\includegraphics[width=.5\textwidth]{linked/map.png}};
        \node [anchor=west] (cdc) at (-1,-5.5) {\footnotesize Map (symbolic) from \cite{russellJones+08a}, H3N2 HA epitope site from \cite{koelBurke+13}};

      \end{tikzpicture}

      \vfill
      Epidemiology: Geographical predictor?\\[2mm]

      Molecular Biology: Is there a telltale mutation site?\\[2mm]

      Immunology: Hemagglutinin inhibition cartography\\[2mm]

      \Large Vision: Bring these levels together
    \end{frame}




    \subsection{The Hemagglutination Inhibition Assay}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{linked/hemagglutination.pdf}
        \caption{Red blood cells (RBC) precipitate.}
        \label{1}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{linked/hemagglutinationAssay.pdf}
        \caption{Influenza Hemagglutinin (HA) coagulates the RBC, forming a mat.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.7\textwidth]{linked/hemagglutinationInhibitionAssay.pdf}
        \caption{Antisera that fit the HA's epitope site bind to it, letting the RBC sink to the bottom. Effect works up to a certain antigenic distance and antiserum concentration.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=.9\textwidth]{linked/HIassaylab1.pdf}
        \caption{\footnotesize Test 12 different dilutions of one antiserum against 8 different virus strains. The highest dilution that prevents agglutination is called the titer.}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{How this used to be looked at}


      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-2.9cm,yshift=-5.4cm] at (current page.north east) {\includegraphics[width=.3\textwidth]{linked/hi_smith_map.jpg}};
        \node[xshift=3.9cm,yshift=-4.5cm] at (current page.north west) {\includegraphics[width=.5\textwidth]{linked/HItable.png}};
        \node [anchor=west] (cdc) at (-.8,-2.5) {\footnotesize Table from \cite{bedford15},};
        \node [anchor=west] (cdc) at (-.8,-3.2) {\footnotesize cartography from \cite{smithLapedes+04}};
        \node [anchor=west] (cdc) at (-.8,-3.6) {\footnotesize \hspace*{.3cm} (one gridline is one $\log_2$ titer i.e. two-fold dilution)};

      \end{tikzpicture}
      % \begin{figure}
      %   \centering
      %   \begin{subfigure}{.5\textwidth}
      %     \centering
      %     \includegraphics[width=\linewidth]{linked/HItable.png}
      %     \caption{A subfigure}
      %     \label{fig:sub1}
      %   \end{subfigure}%
      %   \begin{subfigure}{.5\textwidth}
      %     \centering
      %     \includegraphics[width=.4\linewidth]{linked/hi_smith_map.jpg}
      %     \caption{A subfigure}
      %     \label{fig:sub2}
      %   \end{subfigure}
      %   \caption{A figure with two subfigures}
      %   \label{fig:test}
      %   \end{figure}
    \end{frame}

    \subsection{Mapping Antigenicity to the Tree}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/HItableToTree.pdf}
        \caption{\footnotesize Mapping the chart to the tree inferred from sequences. \cite{bedford15}}
      \end{figure}
      \addtocounter{figure}{-1}

    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/HItableToTreeDiffs.pdf}
        \caption{\footnotesize Mapping the chart to the tree inferred from sequences. \cite{bedford15}}
      \end{figure}
    \end{frame}


    \begin{frame}{\subsecname}
      \framesubtitle{\small A literal toy model:}
      \begin{figure}
        \includegraphics[width=.58\textwidth]{linked/shapesOfWood.jpg}
        \caption{\footnotesize \cite{rosipaw10}}
      \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]{The Model --- \cite{neherBedford+16} ---}
      \framesubtitle{}
      $T_{a\beta}~\ $........................... HI titer of virus $a$ against antiserum $\beta$ (virus $b$)

      $H_{a\beta}~$........................... $\log_2$ relative titer (we'll use this one)
      \begin{align}
        H_{a\beta} = \log_2 (T_{b\beta}) ~ - ~\log_2 ( T_{a\beta})
      \end{align}

      $\hat{H}_{a\beta}~$........................... predicted $\log_2$ relative titer
      $v_{a}~~~$........................... avidity of virus $a$ (=greediness)
      $p_{\beta}~~~$........................... potency of antiserum $\beta$ (=effectiveness)
      $D_{a\beta}~$........................... genetic component of titer drop

      \begin{align}
        \hat{H}_{a\beta} = v_a + p_\beta + D_{ab}
      \end{align}


      What remains is split up into a sum over individual branch contributions $d_i \ge 0$:
      \begin{align}
        D_{ab} = \sum_{i \in (a ... b)} d_i
      \end{align}
      Where the sum is over the path connecting virus $a$ and virus $ b $ corresponding to antiserum $\beta$. Now we want
      \begin{align}
        \hat{H}_{a\beta} \stackrel{!}={} {H}_{a\beta}
      \end{align}
      to that end we minimize a cost function $C$ of the whole tree:
      \begin{align}
        C := \sum_{a,\beta}( \hat{H}_{a\beta} - {H}_{a\beta})^2 + \lambda \sum_i d_i + \gamma \sum_a v_a^2 + \delta \sum_\alpha p_\alpha^2
      \end{align}

    \end{frame}














  \subsection{Predictions}



    \begin{frame}{\subsecname}
      \framesubtitle{of antigenicity}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/neher+16/tree_model.pdf}
        \caption{\footnotesize On the left, 10\% randomly picked measurements were omitted,\

        on the right, 10 \% of entire titer columns were held back, as if it was a new clade.\

        Dataset: Influenza A/H3N2 (12y) \cite{neherBedford+16}}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{corroborating "tree-likeness"}
      \begin{figure}
        \includegraphics[width=.8\textwidth]{linked/neher+16/tree_model_hist.pdf}
        \caption{\footnotesize Left: Does subtracting $v_a$ and $p_\beta$ enforce symmetry? ($\Delta_{a\beta} \stackrel{\cdot}={} D_{ab}$)\
        
        Right: showing tree-likeness, employing the quartet rule. \cite{neherBedford+16}}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{fixation implications}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/neher+16/recent_antigenic_evolution.pdf}
        \caption{\footnotesize Fraction of samples having a certain mutation plotted over time.

        Strains with frequencies smaller than $0.01$ were omitted. \cite{neherBedford+16}}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{recall:}
      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-4cm,yshift=-1.5cm] at (current page.north east) {\includegraphics[width=.7\textwidth]{linked/neher+16/recent_antigenic_evolution.pdf}};
      \end{tikzpicture}
      \begin{figure}
        \includegraphics[width=.65\textwidth]{linked/h3n2escape.pdf}
        \caption{\footnotesize A/H3N2 phylogeny \cite{volzKoelle+13}}
      \end{figure}
    \end{frame}

    \begin{frame}{\subsecname}
      \framesubtitle{fixation jumps}
      \begin{tikzpicture}[remember picture,overlay]
        \node[xshift=-4cm,yshift=-1.5cm] at (current page.north east) {\includegraphics[width=.7\textwidth]{linked/neher+16/recent_antigenic_evolution.pdf}};
      \end{tikzpicture}
      \begin{figure}
        \includegraphics[width=.8\textwidth]{linked/neher+16/fraction_frequency_threshold_colourbar.pdf}
        \caption{\footnotesize For high recent antigenic evolution traits, $25\%$ prevalence directly entails $75\%$ \cite{neherBedford+16}}
      \end{figure}
    \end{frame}







  \section{\textit{Nextstrain}}

    \begin{frame}

      \begin{figure}
        \includegraphics[width=.8\textwidth]{linked/nextstrain.png}
      \end{figure}
      \begin{tikzpicture}[remember picture,overlay]
        \node [anchor=west] (cdc) at (-.8,-3.6) {\footnotesize \cite{leeMoncla+20}};

      \end{tikzpicture}
    \end{frame}

    \begin{frame}{\secname --- A Growing Platform}
      \begin{figure}
        \includegraphics[width=\textwidth]{linked/NextstrainPathogens.pdf}
        \vspace*{-.2cm}
        \caption{\footnotesize }
      \end{figure}
      \begin{tikzpicture}[remember picture,overlay]
        \node [anchor=west] (cdc) at (-.8,-3.6) {\footnotesize \cite{leeMoncla+20}};

      \end{tikzpicture}
    \end{frame}

    \subsection{Hands-On with the Narrative}

    \begin{frame}{\secname --- \subsecname}
      \framesubtitle{}
      \href{URL}{https://nextstrain.org/community/narratives/gitchhiker/virophyle}
    \end{frame}

    \section{Outlook}

    \begin{frame}{Outlook}
      \framesubtitle{and closing remarks}
      \begin{itemize}
        \item Difficult to disentangle levels, therefore integrate visualization
        \item Include more meta data (symptoms, severity, \& c.)
        \item Interesting: larger evolutionary timescale
        \item Include other pathogens, like bacteria, but careful: HGT
        \item Nice to see instance of Red Queen Hypothesis
        \item The nextstrain workings were very transparent and user friendly.
      \end{itemize}
    \end{frame}

    \begin{frame}[allowframebreaks]{References}
    \tiny
\bibliography{virophyle}
\bibliographystyle{apalike}
%\printbibliography
    \end{frame}




  \end{darkframes}

\end{document}
