\documentclass[12pt]{scrartcl}

\author{Paul Wiesemeyer}
\title{Phylogenetics for Predicting Virus Evolution}
\subtitle{Including a brief guide to the open science tool {\LARGE\textit{nextstrain.org}} \texorpdfstring{\\[1cm]}{,}A Contribution to the 2020 Seminar ``Physics of Viruses'' Held by Ulrich Schwarz and Frederik Graw at University Heidelberg }
\date{\today}


\input{latex-includes/includes}
\usepackage[square]{natbib}

\usepackage{tikz}
\usetikzlibrary{trees}
\usetikzlibrary{shapes}
\usepackage{amsmath}
\usepackage{xspace}
\usepackage{forest}
\usepackage{hyperref}
\usepackage{graphicx}

\usepackage[acronym]{glossaries}
\renewcommand{\glstextformat}[1]{\textit{#1}}
\makeglossaries
\include{latex-includes/glossary}


\newcommand{\red}[1]{{\color{red}#1}}



\begin{document}

\maketitle

\section*{Aim of this write-up}

  After reading these pages\textemdash and potentially following the recommended material at the end of each section\textemdash the reader will be able to:
  \begin{itemize}
    \item Read phylogenetic trees
    \item Explore the framework nextstrain.org
    \item Have some in-detail knowledge about influenza
    \item Have gotten a close look at the phylogenetic approach at prediction of viral evolution
    \item ... and hopefully end up with a sprawling interest in virus research.
  \end{itemize}

  As the research on this vast field proved cumbersome due to many terms and definitions unfamiliar to a physics graduate, this write-up is augmented by a Glossary and List of Abbreviations (at the very end).
\tableofcontents

\section{Introduction}

  \subsection{Problem at Hand}

    Viruses, and the diseases they provoke, put a large burden on human society. \red{Viruses are ...(TODO) .} comparably simple structures are capable of modifying and possibly destroying vital functions in the human body, by infiltrating their genetic material into the reproductory apparatus of cells.

    All viruses that persist in the human population over longer times share some common features:
    \begin{itemize}
      \item They need some form of protection from their surroundings, a hull
      \item They need to find a way to get into human cells
      \item They aim at reproducing quickly
      \item If there is no constant infection source, they need to find a path to get from one individual to another
      \item They will have to deal with human immune system response
      \item They will underlie some kind of evolutionary pressure (too general).
    \end{itemize}

    (This rough characterization is of course incomplete, and there may also be exceptions. The terms ``need'' and ``aim at'' are to be understood as: evolutionary processes strongly favor these characteristics.)

    To give another rough image of what makes up a virus, here are the essential constituents:
    \begin{itemize}
      \item A piece of genetic code, most importantly categorized into: \acrfull{rna} or \acrfull{dna}, single or double stranded, positive or negative sense, length (usually some few kilo base (pairs) long)
      \item A hull: either only a capsid (of proteins) or an additional envelope (bilipid layer).
    \end{itemize}

    For most enveloped viruses such as members of the \textit{Orthomyxoviridae} family (including influenza) or \textit{Coronaviridae} (e.g. SARS-CoV-2), there is a variety of different surface proteins that populate the bi-lipid layer.

    These surface proteins take up functions such as binding to a cell to infiltrate the viruses genetic sequence into it, or releasing a freshly assembled virus from the host cell's surface into the surrounding body fluids. This is usually done via interactions with specific host cell receptors, therefore defining a \acrfull{rbd} as the part of the surface protein that fits onto the receptor like a key fits into a lock.

    At the same time, the human immune system will also interact with mostly surface proteins. The immune system will eventually develop antibodies that are targeted to bind to a specific region of virus surface proteins, thereby rendering them innocuous. The surface protein region targeted by the immune system is called \textit{epitope site} and may have a large overlap with the RBD.

    These two mechanisms put an evolutionary pressure on the virus, especially on the surface proteins and its epitope sites. While having to maintain essential functioning such as binding to receptors, the virus will draw large advantage from modifying its epitope site to such an extent that the antibodies cannot bind to it any longer. This way of disguising itself\textemdash by amino acid mutations in crucial places\textemdash will allow the virus to reinfect previously immune individuals of the host population.

    An accelerated evolution therefore allows viruses to persist in the human population over long periods, but at the same time provides a large record to trace its spreading history, when the viral genome is sequenced and the information is curated.

    In this write-up, we will look at how sequencing of the viral genome, along with phylogenetic tree inference, can provide useful insights on (i) the route that the virus takes to spread in the human population, (ii) the prevalence of strains within a virus type, and (iii) the (projected) evasion of the virus from immune system response and vaccines.

    \red{Vaccines need updates. Scientific publishing and vaccine manufacturing take in the order of half a year each. Two influenza seasons (at a given place)are usually only a few months apart.}

    The goal of the \textit{nextstrain} research group is to make phylogenetic trees inferred from external sequencing labs quickly accessible and easily explorable for fellow scientists, health care officials and the public. This is urgently needed in highly dynamic situations such as the 2020 COVID-19 pandemic. Also, \textit{nextstrain} provides an integrated visualization of pathogen spreading phenomena, as will be explained in section \ref{nextstrain}.

    Here, we chose to look at influenza, as it is the evergreen. \red{Recombination pathway for evo.}

\section{Phylogenetics}

  Essentially the Wikipedia Article + Volz 2013 (TODO read)

  \subsection{Idea}

  \Gls{phylogenetics} is the

  \subsection{How to Read a Phylogenetic Tree}

  \subsection{Recommended Material}




\section{Influenza}

  Influenza is one of the most common viral diseases \citep{DudaMenna20} and puts a large burden on human health. The disease is estimated to cause symptomatic illness in about $3 - 11 \%$ of the human population (in the U.S.) \citep{tokarsOlsen+18} every season. Globally, up to $650,000$ individuals die from influenza illness every season \cite[see][]{iulianoRoguski+18}.

  Influenza refers to a disease caused by a virus of the influenza family \textit{Orthomyxoviridae}. All \textit{orthomyxo} viruses have an envelope that carries surface proteins and their genome is a negative sense RNA.

  Most relevant for human infectious diseases are the genera \textit{Alpha-} and \textit{Betainfluenzavirus} that contain the species \textit{Influenza A} and \textit{Influenza B} respectively.

  They are further subdivided into their so called \textit{Serotype}, a classification by coagulation behavior in the \gls{hemagglutinationAssay}\\[0mm]


\begin{tabular}{ c c c c c }

Taxonomy: & \hspace{18mm}Family\hspace{13mm} & Genus\hspace{3mm} & Species\hspace{10mm} & Serotype\hspace{1mm}\\
  
\end{tabular}\\[0mm]

\begin{forest}
  for tree={
    grow'=east,
    draw,
    calign=first,
    font=\sffamily,
    rounded corners,
    parent anchor=east,
    child anchor=west,
    edge path={%
      \noexpand\path [\forestoption{edge}] (!u.parent anchor) -- ++(0pt,0) |- (.child anchor)\forestoption{edge label};
    }
  },
  gray/.style={
    draw=none,
    color=gray
  }
  [{Viruses}
    [{$\cdot\cdot\cdot$}, gray
      [{\textit{Orthomyxoviridae}},for tree={l sep= 2mm}
        [{$\alpha$},for tree={l sep= 12mm}
          [{A},for tree={l sep= 18mm}
            [{H1N1}]
            [{H3N2}]
          ]
        ]
        [{$\beta$},for tree={l sep= 12mm}
          [{B},for tree={l sep= 18mm}
            [{Yamagata}]
            [{...}, gray]
          ]
        ]
        [{... (5 more)}, gray]
      ]
      [{\textit{... (149 more)}}, gray, for tree={l sep= 2mm}]
    ]
  ]
\end{forest}

\red{Recombination evo pathway. E.g. Avian H7N9 Flu CFR 15-60 \% but no H2H}


    \subsubsection{Basics}

      History diagram from \citep{alberts15}

      HA: epl

      NA: expl

      Phylo tree of A/H3N2 HA Volz+2013

    \subsubsection{Hemagglutination Inhibition Assay}

      Short description

      Hirst 1943

      mention cartography from Smith+2004 (?)

\section{Nextstrain} \label{nextstrain}

  \subsection{Idea}

  \subsection{Short Usage Instructions}
  \begin{figure}[h!]
    \makebox[\textwidth][c]{
    \includegraphics[width=1.2\textwidth]{linked/screen_red_boxes_next_flu_2y.pdf}}
    \caption{\footnotesize Taken from the \href{https://nextstrain.org/flu/seasonal/h3n2/ha/2y?p=grid}{2 year seasonal H3N2 flu HA dataset visualization} on \href{https://nextstrain.org}{\textbf{nextstrain.org}}, \cite{leeMoncla+20}}
  \end{figure}


  \subsection{Example}

  \subsection{Recommended Material}

\section{Mapping Influenza Evolution}

  \subsection{Mapping Titer to Tree}

    Minimizing a Cost function

    Tree model vs. substitution model

    Proving Treelikeness

  \subsection{Results}

  \begin{figure}[h!]
    \includegraphics[width=\textwidth]{linked/neher+16/recent_antigenic_evolution.pdf}
    \caption{\footnotesize For high recent antigenic evolution traits, $25\%$ prevalence directly entails $75\%$ \cite{neherBedford+16}}
  \end{figure}

  \begin{figure}[h!]
    \includegraphics[width=.8\textwidth]{linked/neher+16/fraction_frequency_threshold_colourbar.pdf}
    \caption{\footnotesize For high recent antigenic evolution traits, $25\%$ prevalence directly entails $75\%$ \cite{neherBedford+16}}
  \end{figure}

    Figure 6 Interpretation

\section{Conclusion}

  Veryasfsadffds interestidsfng! \gls{maths} \acrfull{gcd} sdfg \gls{latex}

\clearpage

\section{References, Acronyms, and Glossary}


\bibliography{virophyle}
\bibliographystyle{apalike}

\printglossary[type=\acronymtype]

\printglossary

% \include{latex-includes/introduction}

% \include{latex-includes/influenza}

% \clearpage

% \section*{Appendix}





\end{document}
