\documentclass[12pt]{scrartcl}

\author{Paul Wiesemeyer}

\title{Phylogenetics\texorpdfstring{\\}{,} for Predicting Virus Evolution}

\subtitle{Including a brief guide to the open science tool {\LARGE\textit{nextstrain.org}} \texorpdfstring{\\[1cm]}{,}A Contribution to the 2020 Seminar ``Physics of Viruses'' Conducted by Ulrich Schwarz and Frederik Graw at University Heidelberg }

\date{\today}


\input{latex-includes/includes}
\usepackage[square]{natbib}

\usepackage{tikz}
\usetikzlibrary{trees}
\usetikzlibrary{shapes}
\usepackage{amsmath}
\usepackage{xspace}
\usepackage{forest}
\usepackage{hyperref}
\usepackage{listing}

\usepackage{graphicx}
\usepackage{lmodern}

\usepackage[acronym]{glossaries}
\renewcommand{\glstextformat}[1]{\textit{#1}}
\makeglossaries
\include{latex-includes/glossary}


\newcommand{\red}[1]{{\color{red}#1}}
\newcommand{\gray}[1]{{\color{gray}#1}}

\glsaddall

\begin{document}

\maketitle

\clearpage

\section*{Aim of this write-up}

  After reading these pages\textemdash and potentially following the recommended material at the end of each section\textemdash the reader will be able to:
  \begin{itemize}
    \item Read phylogenetic trees
    \item Explore the framework nextstrain.org
    \item Have some in-detail knowledge about influenza
    \item Have gotten a close look at the phylogenetic approach at prediction of influenza evolution
    \item ... and hopefully end up with a sprawling interest in virus research!
  \end{itemize}

  As the research on this vast field proved cumbersome due to many terms and definitions unfamiliar to a physics graduate, this write-up comprises a glossary and a list of acronyms at the very end. Included terms are \textit{italicized}.

\vfill

\tableofcontents

\clearpage

\section{Introduction}

  \subsection*{Old Introduction}
    \gray{

      Viruses, and the diseases they provoke, put a large burden on human society. \red{Viruses are ...(TODO) .} comparably simple structures are capable of modifying and possibly destroying vital functions in the human body, by infiltrating their genetic material into the reproductory apparatus of cells.

      All viruses that persist in the human population over longer times share some common features:
      \begin{itemize}
        \item They need some form of protection from their surroundings, a hull
        \item They need to find a way to get into human cells
        \item They aim at reproducing quickly
        \item If there is no constant infection source, they need to find a path to get from one individual to another
        \item They will have to deal with human immune system response
        \item They will underlie some kind of evolutionary pressure (too general).
      \end{itemize}

      (This rough characterization is of course incomplete, and there may also be exceptions. The terms ``need'' and ``aim at'' are to be understood as: evolutionary processes strongly favor these characteristics.)

      To give another rough image of what makes up a virus, here are the essential constituents:
      \begin{itemize}
        \item A piece of genetic code, most importantly categorized into: \acrfull{rna} or \acrfull{dna}, single or double stranded, positive or negative sense, length (usually some few kilo base (pairs) long)
        \item A hull: either only a capsid (of proteins) or an additional envelope (bilipid layer).
      \end{itemize}

      For most enveloped viruses such as members of the \textit{Orthomyxoviridae} family (including influenza) or \textit{Coronaviridae} (e.g. SARS-CoV-2), there is a variety of different surface proteins that populate the bi-lipid layer.

      These surface proteins take up functions such as binding to a cell to infiltrate the viruses genetic sequence into it, or releasing a freshly assembled virus from the host cell's surface into the surrounding body fluids. This is usually done via interactions with specific host cell receptors, therefore defining a \acrfull{rbd} as the part of the surface protein that fits onto the receptor like a key fits into a lock.

      At the same time, the human immune system will also interact with mostly surface proteins. The immune system will eventually develop antibodies that are targeted to bind to a specific region of virus surface proteins, thereby rendering them innocuous. The surface protein region targeted by the immune system is called \textit{epitope site} and may have a large overlap with the \acrshort{rbd}.

      These two mechanisms put an evolutionary pressure on the virus, especially on the surface proteins and its epitope sites. While having to maintain essential functioning such as binding to receptors, the virus will draw large advantage from modifying its epitope site to such an extent that the antibodies cannot bind to it any longer. This way of disguising itself\textemdash by amino acid mutations in crucial places\textemdash will allow the virus to reinfect previously immune individuals of the host population.

      An accelerated evolution therefore allows viruses to persist in the human population over long periods, but at the same time provides a large record to trace its spreading history, when the viral genome is sequenced and the information is curated.

      In this write-up, we will look at how sequencing of the viral genome, along with phylogenetic tree inference, can provide useful insights on (i) the route that the virus takes to spread in the human population, (ii) the prevalence of strains within a virus type, and (iii) the (projected) evasion of the virus from immune system response and vaccines.

      \red{Vaccines need updates. Scientific publishing and vaccine manufacturing take in the order of half a year each. Two influenza seasons (at a given place)are usually only a few months apart.}

      The goal of the \textit{nextstrain} research group is to make phylogenetic trees inferred from external sequencing labs quickly accessible and easily explorable for fellow scientists, health care officials and the public. This is urgently needed in highly dynamic situations such as the 2020 COVID-19 pandemic. Also, \textit{nextstrain} provides an integrated visualization of pathogen spreading phenomena, as will be explained in section \ref{nextstrain}.

      Here, we chose to look at influenza, as it is the evergreen. \red{Recombination pathway for evo.}

    } % grey

    \subsection{What are viruses?}

    History: infectious fluids w/o bacteria. Turns out the mere information can modify a living being, using its reproductory apparatus. Viruses also mark the border to what is considered the living world, being classified as below it.

    Functional definition: Genetic material. Mechanism to enter ``hijack'' host cell. Often: protective hull.

    Detailed definition (box?): Different types of genetic code. Different types of protection from the outer world. Different mechanisms to enter a cell.

    Include: Spreading of viruses. Airborne, Waterborne etc. most importantly: directly host-to-host, or does it need a source or intermediary host or reservoir?

    Organizational structure of human society makes up a big threat when it comes to human-to-human transmission. What used to be a spatial problem in past centuries (e.g. the wavelike patterns by the bubonic plague) has now turned into a highly interconnected world, that is the structure of the population has changed dramatically. This accelerates pathogen transmission, as it ... but it also open new opportunities for countermeasures.

    \subsection{What is problematic when viruses evolve?}

    Evade immune system. And even vaccines. Leads to a race and so called co-evolution. Mention the Red Queen here.

    But one big advantage: We can now (since when?) sequence the genetic material and through the statistical nature of its modifications (mutations) extract information from it.

    Most importantly we can interrelate the different probes. Where we just had 100 people with same symptoms, we can now make precise statements about who probably got the virus from whom. With a certain uncertainty.

    So how can we make sense of this?

    The theory behind this is called phylogenetics.



\section{Phylogenetics}

  The classical example is that you have a number of different genetic sequences. Could be from different species or different individuals of the same species, as we will do here.

  Then you need a model about how nature gets from one instance of a sequence to the other. The usually assumed manner is by point mutations and indels. And that they happen stochastically.

  Picture of two different trees that explain three (or four?) different histories of a sequence.

  Parsimony: Number of needed substitutions is to be reduced. This makes especially sense if single mutation probabilities are low. It is Ockham's razor applied.

  \textbf{Molecular clock}: If the substitution rate was constant over all sites of the genome, one would expect the stochastically occurring mutations to yield one constant mutation rate for the whole genome. In reality, this is a strong simplification and each site has their own mutation rate. This is due to different evolutionary pressure laying on each site.

  Synonymous mutations are such that do \textit{not} alter the expressed amino acid. They leave the organism unchanged and are so to say just for our pleasure (??). Non-synonymous mutations change the amino acid and are therefore under selective pressure. That is: depending on whether they are beneficial or detrimental to the organism, they will establish in the population or vanish and be unlikely to show up in any probes. That is the nature of evolution, that---on large timescales---we only see in the data what is capable of surviving and reproducing.

  \subsection{What defines a Phylogenetic Tree?}

  Topological construct: Vertices linked by edges. To begin with, lengths and shapes have no meaning, only connectedness is relevant.

  A construct of vertices and edges (called graph or network) that has no internal loops. Consequently, if you remove one edge of a tree, it decays into two disconnected graphs.

  Vertices or nodes can be divided into external and internal, the external ones are sometimes called leaves.

  Furthermore, trees are usually taken to be bifurcating, that is one internal vertex has exactly three edges. Trees that do not fullfill this criterion are called multifurcating or polytomous, which can then be looked at as a lack of resolution, as a multifurcating vertex can always be reduced to a set of bifurcating ones.

  Now the idea is that you take a bunch of external nodes (sequences) and reconstruct their interrelatedness in form of such a tree by probabilistic considerations. This is called tree inference and more info can be found HERE.

  In the next step, one would like to give the vertices an order, as in who are the descendants of whom. This is done by introducing a special vertex called root, which is then parent to all other vertices. Be it pointed out here, that it has nothing to do with time, so far we are talking only topology and hierarchy.

  For N leaves there are $\frac{(2N - 3)!}{2^{N-2}(N-2)!}$ possible bifurcating rooted trees (Wikipedia).

  \subsection{Visualization of especially sequence data?}

  One dimension could be mutation events or time

  The other has no meaning.

  Radial trees are only a way of clamming more information into a 2D piece of paper, as opposed to Cartesian visualizations.

  PICTURE?

  \subsection{Phylodynamics and what we can read from a tree}

  Phylodynamics---as opposed to phylogenetics---is not satisfied with a tree but also asks: What processes shape a viral phylogeny?

  First: Data availability. As long as scientific field does not abound with data, or as data content is dominated by the means available to gather it, a phylogenetic tree will always rather be the skewed little window through which we look at a problem, than an accurate image of any of the processes below.

  Second: Epidemiological processes. For example growth vs. constant population PICTURE.

  Third: Immunological processes. For example imbalanced vs. balanced. Maybe HIV as a PICTURE here? Also included here: Evolutionary feedbacks.

  Fourth: Host population structure.

  Note, that we are increasingly adding information to the vertices, leaving the theoretical realm of tree construction and entering into the matter of interpreting trees that have been constructed by appropriate algorithms.

  As a last note, the mapping between these processes and the observed ML tree is many-to-one REF VOLZ+2013, that is, there are usually many possible explanations for one observation, and phylodynamics deals with this interplay.

  \subsection{Recommended Material}

  Good Phylogenetics page

  Anders Gorm Petersen

  Volz+2013

  But---to continue our journey through phylodynamic methods---let us now take a closer look at one most interesting virus family, the influenza viruses.


\section{Influenza}

  Influenza is one of the most common viral diseases \citep{DudaMenna20} and puts a large burden on human health. The disease is estimated to cause symptomatic illness in about $3 - 11 \%$ of the human population (in the U.S.) \citep{tokarsOlsen+18} every season. Globally, up to $650,000$ individuals die from influenza illness every season \cite[see][]{iulianoRoguski+18}.

  Influenza refers to a disease caused by a virus of the influenza family \textit{Orthomyxoviridae}. All \textit{orthomyxo} viruses have an envelope that carries surface proteins and their genome is a negative sense RNA.

  Most relevant for human infectious diseases are the genera \textit{Alpha-} and \textit{Betainfluenzavirus} that contain the species \textit{Influenza A} and \textit{Influenza B} respectively.

  They are further subdivided into their so called \textit{Serotype}, a classification by coagulation behavior in the \gls{hemagglutinationAssay}\\[0mm]


  \begin{tabular}{ c c c c c }

   Taxonomy: & \hspace{18mm}Family\hspace{13mm} & Genus\hspace{3mm} & Species\hspace{10mm} & Serotype\hspace{1mm}\\
    
  \end{tabular}\\[0mm]

  \begin{forest}
    for tree={
      grow'=east,
      draw,
      calign=first,
      font=\sffamily,
      rounded corners,
      parent anchor=east,
      child anchor=west,
      edge path={%
        \noexpand\path [\forestoption{edge}] (!u.parent anchor) -- ++(0pt,0) |- (.child anchor)\forestoption{edge label};
      }
    },
    gray/.style={
      draw=none,
      color=gray
    }
    [{Viruses}
      [{$\cdot\cdot\cdot$}, gray
        [{\textit{Orthomyxoviridae}},for tree={l sep= 2mm}
          [{$\alpha$},for tree={l sep= 12mm}
            [{A},for tree={l sep= 18mm}
              [{H1N1}]
              [{H3N2}]
            ]
          ]
          [{$\beta$},for tree={l sep= 12mm}
            [{B},for tree={l sep= 18mm}
              [{Yamagata}]
              [{...}, gray]
            ]
          ]
          [{... (5 more)}, gray]
        ]
        [{\textit{... (149 more)}}, gray, for tree={l sep= 2mm}]
      ]
    ]
  \end{forest}

  Where they come from (mention important pandemics).

  INFLUVENNZA PICTURE

  \subsection{Hemagglutination Inhibition Assay}

    Hemagglutinin is the surface protein for entering host cells. It is intended to dock at the XXX receptor, but can also bind to red blood cell's YYY which is where it got its name from.

    When a virus is added to blood serum, the virus and RBCs will form a matrix structure, the blood will agglutinate. This is particularly useful since it is an effect visible with bare eye.

    In Hemagglutination Inhibition Assays, a third component is added: Antibodies to a particular influenza subtype that were incubated in for example eggs. These antibodies bind to the virus epitope sites, preventing the blood from coagulating. This process depends on the antibody concentration and how closely related they are

    Hirst 1943

    mention cartography from Smith+2004 (?) 

  \subsection{What is special about influenza?}

    It has a relatively high mutation rate (EXACT COMPARISON, PLOT?), that is X substitutions per kilo base per reproductory cycle or per year (distinguish here to evolutionary rate?). 
    Recombination evo pathway. E.g. Avian H7N9 Flu CFR 15-60 \% but no H2H


  \subsection{Basics}

    History diagram from \citep{alberts15}

    HA: epl

    NA: expl

    Phylo tree of A/H3N2 HA Volz+2013


\section{Nextstrain} \label{nextstrain}

  In 2015, a group of scientists from Fred Hutchinson Cancer Research Center and University of Basel published a web tool to track and visualize the evolution of influenza viruses. Under the name \textit{nextflu}, they had automatized sequence alignment, tree inference and visualization into two data processing pipelines, allowing for a quick analysis of the genomic data at hand \citep{neherBedford15a}.

  Shortly thereafter, phylodynamic analyses of other viruses like \glsdisp{zika}{zika} and \glsdisp{ebola}{ebola} were included in the platform, now running under the name nextstrain.

  Realizing that a limiting factor of phylodynamics is data availability, the aim was to make the sharing of sequence data and phylodynamic analyses easier between scientists. At the same time nextstrain allowed to directly present the phylogenetic results to the scientific community and the public.

  In February 2017 the team around nextstrain won the first \gls{openScience} prize for their work \citep{hudson17}.

  While bringing the scientific community and the public closer together, the ingenuity of nextstrain can as well be found in another domain: The integration of the multiple scales of phylodynamic processes into one visualization. So far, evolution of the viral genome had been studied by looking at the phylogeny while epidemic spreading patterns were visualized for example with geographic maps of the affected regions. Nextstrain provides an integrated view through these two different windows being in sync, allowing for a picture much closer to the process found in nature.

  The following section will give a short hands-on introduction to the platform and its most relevant features.

  \subsection{Using nextstrain}

  On \textit{nextstrain.org}, the visitor can first chose between a wide range of available pathogens. Beside viruses, even a bacterial disease---tuberculosis---can be selected. Contributions by external scientific groups are included further down the main page.

  \begin{figure}[h!]
    \makebox[\textwidth][c]{
    \includegraphics[width=.8\textwidth]{linked/NextstrainPathogens.pdf}}
    \caption{Augmented pathogen visualization, modified from \href{https://nextstrain.org}{\textbf{nextstrain.org}}, \cite{leeMoncla+20}}
  \end{figure}

  Selecting for instance seasonal flu, the visitor gets to what we will call the \textit{phylodynamic dashboard}. In the following, we will go through its most basic features and discuss some of its possible uses.

  The dashboard consists of primarily a phylogeny and a geographic map, as well as a diversity and frequency panel further down.

  \begin{figure}[h!]
    \makebox[\textwidth][c]{
    \includegraphics[width=1.2\textwidth]{linked/screen_red_boxes_next_flu_2y.pdf}}
    \caption{\footnotesize Taken from the \href{https://nextstrain.org/flu/seasonal/h3n2/ha/2y?p=grid}{2 year seasonal H3N2 flu HA dataset visualization} on \href{https://nextstrain.org}{\textbf{nextstrain.org}}, \cite{leeMoncla+20}}
  \end{figure}

  The \textbf{Phylogeny} may be considered the core of the visualization. It displays sequences as colored dots, showing the collected meta data when hovering or clicking. The branches are of the inferred \acrshort{ml} tree, and come with a confidence interval that is displayed when hovering. The black crosses are exclusive to the influenza datasets, they mark the location of a vaccine strain.

  The \textbf{Geographic Map}, shows the same data as the phylogeny, but mapped out according to the sequencing location (from meta data). The coloring is the same as the phylogeny. Some datasets display transmission lines between the geographic regions, these are a reflection of the branches of the phylogeny.

  The \textbf{play button} allows to watch the disease unfolding as an animation, where the phylogeny and the geography is in sync.

  Below, the \textbf{Diversity Panel} shows the sequence under investigation, with an identification of specific regions such as the signal peptide. The histogram displays either the number of mutation events or the entropy of each site in the genome. Entropy is calculated from the presence of different mutations $ S = -\sum_{\text{mutations } i}(p_i \log(p_i))$, where $p_i$ is the probability that a randomly picked sequence shows the mutation $i$ at the site, such that exactly two equally abundant mutations would yield an entropy of about $.69$. The histogram can be binned by \acrfull{aa} or by \acrfull{nt}. If a genome site is clicked, the entire dashboard coloring will be set to discriminate between different mutations of that very position.

  The \textbf{Frequency Panel} displays the abundance of the selected variation over the time period of the dataset and a projection beyond. Here, \textit{``selected variation''} refers to any information that the data can be colored by, be it the strain (as is the default) or a specific mutation or anything else. For example when coloring by a high entropy site, the frequency panel should display at least two abundant different colors. The \textbf{Projection} of the frequencies is a feature specific to the influenza dataset, and its scientific origin will be discussed in section \ref{prediction}.

  In the leftmost \textbf{navigation bar}, the main visualization adjustments can be made. Primarily, the dataset can be narrowed down to a specific part of the genome, or to a time range over which the data was collected. The data usually originates from many laboratories throughout the world submitting their sequences





  \subsection{Example}

  WILL I DO THIS?

  \subsection{Recommended Material}

    The platform is self-explanatory to some degree, but for a deeper introduction to the topic it provides plenty of material:
\begin{itemize}
  \item The Documentation and Help. It should answer the most frequent questions and it gives a detailed guide on how to build an entire phylodynamic analysis on your own machine with the pipelines \verb|augur| and \verb|auspice|.
  \item The Discussion forum

  \item Guide on Narratives
\end{itemize}

  The video series with Colin Megill

\section{Mapping Influenza Evolution} \label{prediction}

  In this section, we will look at how the prediction of future influenza strain prevalence has been done by \cite{neherBedford+16}. We will begin by mapping antigenicity to the phylogeny and then end in a statement about the correlation between recent antigenic advancement and the probability to reach certain frequency thresholds.

  \subsection{Mapping Titer to Tree}

    The dataset that we start with will have to contain the following data:
    \begin{itemize}
      \item The set of sequences of the viral genome (with the given meta data)
      \item The set of antiviral vaccine strains
      \item A table of \glspl{titer} values between antiviral and viral strains
      \item The inferred phylogenetic tree of the viral sequences (can be produced with nextstrain's pipeline \verb|augur| or other \textit{off the shelf} tree inference algorithms)
    \end{itemize}

    To start off with, we want to normalize the \glspl{titer} to the surplus of twofold dilutions to the reference virus. Such that if $T_{a\beta}$ is the titer between a virus $a$ and an anti virus $\beta$, we have

    \begin{align}
      H_{a\beta} = \log_2T_{b\beta} - \log_2T_{a\beta}
    \end{align}

    where $b$ is the virus that anti virus $\beta$ was intended for, that is $T_{b\beta}$ is the so called \textit{homologous} titer.

    Now we want to model this data table as an outcome of one of the two following processes:
    \begin{itemize}
      \item the bifurcation of the tree into ever smaller branches
      \item the substitution of amino acids that can be read directly from the genome.
    \end{itemize}

    For that, we set up two different models, one \textit{tree model} that assigns an antigenic change $d_i$ to each internal node (bifurcation) of the tree, and another model called \textit{substitution} model, that lets each \textit{non-silent} substitution in the sequence produce an antigenicity change $d_i$.

    For the \textit{modeled} standardized titers we write down
    \begin{align}
      \hat{H}_{a\beta} = v_a + p_\beta + \sum_{i\in(a...b)}d_i
    \end{align}

    where the hat means ``modeled'', $v_a$ refer


    Minimizing a Cost function

    Tree model vs. substitution model

    Proving Treelikeness

  \subsection{Results}

  \begin{figure}[h!]
    \includegraphics[width=\textwidth]{linked/neher+16/recent_antigenic_evolution.pdf}
    \caption{\footnotesize For high recent antigenic evolution traits, $25\%$ prevalence directly entails $75\%$ \cite{neherBedford+16}}
  \end{figure}

  \begin{figure}[h!]
    \includegraphics[width=.8\textwidth]{linked/neher+16/fraction_frequency_threshold_colourbar.pdf}
    \caption{\footnotesize For high recent antigenic evolution traits, $25\%$ prevalence directly entails $75\%$ \cite{neherBedford+16}}
  \end{figure}

    Figure 6 Interpretation

\section{Conclusion}

  Veryasfsadffds interestidsfng!

\clearpage


\section{References, Acronyms, and Glossary}


\bibliography{virophyle}
\bibliographystyle{apalike}

\printglossary[type=\acronymtype]

\printglossary

% \include{latex-includes/introduction}

% \include{latex-includes/influenza}

% \clearpage

% \section*{Appendix}





\end{document}
