---
title: "Welcome to the test narrative for the Physics of Viruses seminar. We will get familiar with most of the panels on the right side. Please HOVER YOUR MOUSE HERE and scroll down, or press the down arrow."
authors: "Paul Wiesemeyer"
affiliations: "pwieseme@iup.uni-heidelberg.de"
date: "June 23, 2018"
dataset: "https://nextstrain.org/flu/seasonal/h3n2/ha/12y"
abstract: ""
---


# [On the left you can see the phylogenetic tree, the corresponding map, a divergence histogram and a frequency diagram.](https://nextstrain.org/flu/seasonal/h3n2/ha/12y?branchLabel=none&p=grid)

The dataset is the hemagglutinin (HA) genome of seasonal H3N2 Influenza (type A).

You can always click on the top right button "EXPLORE THE DATA YOURSELF", and get back to this narrative easily via the same button.

Now let us go through the panels one by one.



# [The Phylogenetic Tree](https://nextstrain.org/flu/seasonal/h3n2/ha/12y?d=tree&p=full)

The phylogenetic tree is the backbone of the whole nextstrain research.

It is inferred via sequence alignment and Bayesian maximum likelihood (ML) methods.

This means, that the dots are known data points, whereas the internal branches come with a confidence interval. Try hovering a branch or a node!

The coloring is by different clades, the Xs show vaccine strains.


# [The Map](https://nextstrain.org/flu/seasonal/h3n2/ha/12y?d=tree&p=full)

Nextstrain's sequences come with metadata, such as the location and date of the sample.


This can be used to create a map, but careful: This does NOT show the number of cases in a certain area. It shows the *number of sampled genomes*, that is it makes a statement about data availability.