---
title: "Welcome to the test narrative for the Physics of Viruses seminar.\n We will get familiar with most of the panels on the right side.\n Please HOVER YOUR MOUSE HERE and scroll down, or press the down arrow."
authors: "Paul Wiesemeyer"
affiliations: "pwieseme@iup.uni-heidelberg.de"
date: "June 23, 2018"
dataset: "https://nextstrain.org/flu/seasonal/h3n2/ha/2y"
abstract: ""
---

<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [On the left you can see the phylogenetic tree, the corresponding map, a divergence histogram and a frequency diagram.](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?branchLabel=none&p=grid)


The dataset is the hemagglutinin (HA) genome of seasonal H3N2 Influenza (type A).
<br><br>
You can always click on the top right button "EXPLORE THE DATA YOURSELF", and get back to this narrative easily via the same button.
<br><br>
Now let us go through the panels one by one.  


<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [The Phylogenetic Tree](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?d=tree&p=full)

The phylogenetic tree is the backbone of the whole nextstrain research. This is our tool to cross-link information and make predictions.
<br><br>
The dots are sequenced data points, whereas the internal branches are ML inferred with respect to topology and length, so they come with a confidence interval. Try hovering a branch or a node!
<br><br>
The coloring is by different clades, the Xs show vaccine strains.
<br><br>
Saturation is used for expressing confidence level, so we end up with 4 dimensions!

<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [The Map](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?d=map&p=full)

Nextstrain's sequences come with metadata, such as the location and date of sampling.
<br><br>
These can be used to populate a map with the data from the phylogenetic tree, but careful:
<br><br>
The map does **not** show the number of cases in a certain area. It shows the _number of sampled genomes_, that is it makes a statement about data availability.

<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [Tree and Map interplay](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?animate=2010-05-23,2020-04-13,1,0,15000&d=tree,map&p=grid)

We can also watch the unfolding of pathogen evolution having the tree and the map in sync.
<br><br>
This is particularly useful for getting a quick overview over a transient situation.
<br><br>

<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [Diversity in the genome](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?d=tree,map,entropy&p=grid)

Very useful is the panel showing the diversity of the genome at each amino acid (AA) or nucleotide position (NT).

Sites of low diversity are a sign for a crucial region in the genome (for example a receptor binding domain).

The genome can be zoomed and, by clicking a particular site the tree is colored according to the corresponding mutations.

<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [Solution](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?c=gt-HA1_145,155,156,158,159,189,193)

We can for example color according to the Koel 7 epitope sites, see [Koel et. al. 2013](https://science.sciencemag.org/content/342/6161/976/tab-figures-data)
<br><br>
The diagram can also be colored with respect to other specific mutations, Antigenic Novelty, Host species, Exposure History e.g. of travelers and much more. (Just explore!)


<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [Frequencies](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?d=tree,map,frequencies&p=grid)


As a last step we can have a look at Frequencies.
<br><br>
They are projected one year into the future and form the most concrete prediction.


<!-- ############ SLIDE BREAK ############# -->
<!-- This is left-side text -->
# [For now, let's go back to the presentation for a conclusion](https://nextstrain.org/flu/seasonal/h3n2/ha/2y?d=tree,map,frequencies&p=grid)


Thanks for making the way around this tiny tutorial and have fun exploring the other pathogens and community contributions!

